app/services/LocationService.js

```diff
@@ -449,4 +453,70 @@ export default class LocationServices {
       // nav.navigate('LocationTrackingScreen', {});
     });
   }
+
+  static getCrossedPaths() {
+    return new Promise(resolve => {
+      GetStoreData(CROSSED_PATHS).then(dayBin => {
+        dayBin = JSON.parse(dayBin);
+        if (dayBin !== null && dayBin.reduce((a, b) => a + b, 0) > 0) {
+          console.log('Found crossed paths');
+          resolve(true);
+        } else {
+          console.log("Can't find crossed paths");
+          resolve(false);
+        }
+      });
+    });
+  }
+  static getParticpating() {
+    return new Promise(resolve => {
+      GetStoreData(PARTICIPATE, false).then(isParticipating => {
+        resolve(isParticipating);
```

Comment:
nit: this will return string `'true'`, so do a `isParticipating === 'true'`

Reply:
It returns a boolean because we pass false as the second argument

Reply:
If it's `'true'` in the store, it will resolve as `'true'` and not `true` though.

Reply:
I am not sure I am following. The getstore will return JSON.parse('true') which returns the boolean true so the promise will return the boolean as well when I resolve it

Reply:
GetStoreData does not `JSON.parse()` if the [value is a string](https://github.com/tripleblindmarket/covid-safe-paths/blob/8e684228a0d543964df402269335470de1d45927/app/helpers/General.js#L14-L16), it really *should*, but it doesn't at the moment.

Reply:
oh. I thought that was isString(value).... nevermind. carry on.